### ISSUE TITLE
Add Pointer-Safety for Safemode.

### ISSUE THREAD

**Issue by assyrianic** [assyrianic/Tagha-Virtual-Machine#26]
Since scripts have an option to enable/disable safemode, I believe safemode should enable pointer-safety at the expense of runtime speed. Safemode can be turned off to increase execution speed but there might be cases where speed isn't a large concern but memory safety is.

This can be done through a few ways.

One way is pushing values like strings, pointers, and arrays will NEED to be entirely copied which will use up an internal heap to store everything.
Something like a vector of chars
```c
vector vecHeap;

// pushing value. Get ptr to string location.
uint8_t *ptr = Tagha_PushString(&vecHeap, "mystring");
union CValue data[] = {
    { .Ptr = ptr }    // send ptr.
};
Tagha_PushValues(tagha, 1, data);
```
Basically, this would achieve memory protection by pushing direct data and using the sure pointer assuming you use it correctly of course!
